qemu/tests/seabios.py

```diff
@@ -34,24 +35,75 @@ def boot_menu_check():
         return (len(re.findall(boot_menu_hint,
                                get_output(seabios_session))) > 1)
 
-    error.context("Start guest with sga bios")
     vm = env.get_vm(params["main_vm"])
     # Since the seabios is displayed in the beginning of guest boot,
     # booting guest here so that we can check all of sgabios/seabios
     # info, especially the correct time of sending boot menu key.
     vm.create()
+    seabios_session = vm.logsessions['seabios']
+    login_timeout = float(params.get("login_timeout", 360))
+    bios_bin_dic = {'rhel6': 'bios.bin', 'rhel7': 'bios-256k.bin'}
+    env_process.preprocess_vm(test, params, env, params.get("main_vm"))
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login(timeout=login_timeout)
+    output = vm.monitor.info("roms")
+    match_bin = re.search(bios_bin_dic['rhel6'], output) or re.search(bios_bin_dic['rhel7'], output)
+    if match_bin:
+        logging.info("Check seabios bin succeeded!")
+    else:
+        err_msg = "Check seabios bin faile"
+        test.error(err_msg)
+    vm.monitor.cmd("system_reset")
+    return
+
+    if params["boot_reboot-timeout"]:
+        rtime_cfg = int(params.get("boot_reboot-timeout"))
+        rtime_sec = rtime_cfg/1000
+        time.sleep(20)
+        if rtime_sec < 60:
+            time.sleep(60)
+        else:
+            time.sleep(rtime_sec)
+
+        output = seabios_session.get_stripped_output()
+        if 0 <= rtime_cfg <= 65535:
+            pattern = "No bootable device.  Retrying in %s seconds" % rtime_sec
+            match1 = re.search(pattern, output)
+            if match1:
+                logging.info("Reboot timeout <65535 setting works!")
+            else:
+                err_msg = "Reboot-timeout tests failed,para is %s" % rtime_sec
+                test.error(err_msg)
+        elif rtime_cfg > 65535:
+            pattern = "No bootable device. Retrying in 65 seconds"
+            match1 = re.search(pattern, output)
+            if match1:
+                logging.info("Reboot timeout >65535 setting works!")
+            else:
+                err_msg = "Reboot-timeout tests failed,para is %s" % rtime_sec
+                test.error(err_msg)
+        elif rtime_cfg < 0:
+            pattern1 = "No bootable device."
+            pattern2 = " Retrying in"
+            match1 = re.search(pattern1, output)
+            match2 = re.search(pattern2, output)
+            if match1 and not match2:
+                logging.info("Reboot timeout <0 setting works!")
+            else:
+                err_msg = "Reboot-timeout tests failed,para is %s" % rtime_sec
+                test.error(err_msg)
+    return
```

Comment:
??

Reply:
removed "return",  This test scenario(rtime_cfg < 0) only need a string "No bootable device."  and  don't need string " Retrying in".